File: kotlin/src/main/kotlin/com/joshy/hywrapper/model/other/CountsResponse.kt
package com.joshy.hywrapper.model.other

import com.joshy.hywrapper.data.GameType
import com.joshy.hywrapper.model.HypixelResponse
import com.joshy.hywrapper.model.RateLimit
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

@Serializable
data class CountsResponse(
    override val success: Boolean? = null,
    override val cause: String? = null,
    val games: Map<String, GameCount>? = null,
    val playerCount: Int? = null,
) : HypixelResponse {
    @Transient
    override var rateLimit: RateLimit? = null

    fun getCountFor(game: GameType): GameCount? {
        return games?.get(game.databaseName) ?: games?.get(game.name)
    }
}

@Serializable
data class GameCount(
    val players: Int? = null,
    val modes: Map<String, Int>? = null,
)

File: python/src/hywrapper/models/other/counts_response.py
from __future__ import annotations

from typing import Dict

from pydantic import BaseModel, ConfigDict, Field

from hywrapper.models.hypixel_response import HypixelResponse


class GameCount(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    players: int = 0
    modes: Dict[str, int] = Field(default_factory=dict)


class CountsResponse(HypixelResponse):
    games: Dict[str, GameCount] = Field(default_factory=dict)
    player_count: int = Field(default=0, alias="playerCount")
